langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java

```diff
@@ -25,21 +27,24 @@
 /**
  * Represents an OpenAI language model, hosted on Azure, that has a chat completion interface, such as gpt-3.5-turbo.
  * <p>
- * Mandatory parameters for initialization are: endpoint, serviceVersion, apiKey and deploymentName.
+ * Mandatory parameters for initialization are: endpoint, serviceVersion, apikey (or an alternate authentication method, see below for more information) and deploymentName.
  * You can also provide your own OpenAIClient instance, if you need more flexibility.
  * <p>
- * There are two primary authentication methods to access Azure OpenAI:
+ * There are 3 authentication methods:
  * <p>
- * 1. API Key Authentication: For this type of authentication, HTTP requests must include the
- * API Key in the "api-key" HTTP header as follows: `api-key: OPENAI_API_KEY`
+ * 1. Azure OpenAI API Key Authentication: this is the most common method, using an Azure OpenAI API key.
+ * You need to provide the OpenAI API Key as a parameter, using the apiKey() method in the Builder, or the apiKey parameter in the constructor.
  * <p>
- * 2. Azure Active Directory Authentication: For this type of authentication, HTTP requests must include the
- * authentication/access token in the "Authorization" HTTP header.
+ * 2. non-Azure OpenAI API Key Authentication: this method allows to use the OpenAI service, instead of Azure OpenAI.
```

Comment:
In this case, I would suggest to:
1. accept `String nonAzureApiKey` in the builder instead of `KeyCredential` for simplicity and also to make it explicit that non-azure service is used
2. provide a default value for `endpoint` = "https://api.openai.com/v1", otherwise it forces user to always specify it